### ISSUE TITLE
[SUGGESTION] Support "dot index" access for tuples as alternative for `std::get`

### ISSUE THREAD

**Issue by bluetarpmedia** [hsutter/cppfront#1027]
Cpp2 could support tuple indexing expressions like [Rust](https://doc.rust-lang.org/reference/expressions/tuple-expr.html#tuple-indexing-expressions) and [Swift](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/expressions#Explicit-Member-Expression) as syntactic sugar for `std::get`.

```cpp
main: () -> int = {
    bag: std::tuple = (1, "test", 3.14);
    std::cout << bag.1 << "\n";         // Instead of std::get<1>(bag)
    return bag.0;                       // Instead of std::get<0>(bag)
}
```

**Will your feature suggestion eliminate X% of security vulnerabilities of a given kind in current C++ code?** 
No

**Will your feature suggestion _automate or eliminate_ X% of current C++ guidance literature?** 
No

**Comment by JohelEGP** [comment]
Seems like something you could implement with Cpp1 reflection (I think it's in the paper).

**Comment by fro0m** [comment]
Looks like [operator[]](https://en.cppreference.com/w/cpp/language/operators) might be an option for accessing tuple members because  in terms of usage, tuple is a collection that can be accessed with indexes.
Also for example in Python the [interface of tuple is similar to list](https://www.geeksforgeeks.org/python-list-vs-array-vs-tuple/) with accessing with [index].

**Comment by bluetarpmedia** [comment]
> The index needs to be known at compile time though, as the return type depends on it.

Yes, the `tup.1` syntax proposed above would be lowered to `std::get<1>(tup)`.

**Comment by jcanizales** [comment]
I know I was replying to the suggestion to use `operator[]`. Although maybe such a member function can be written with Cpp2's existing tools?
